Lib/test/test_time.py

```diff
@@ -511,7 +516,10 @@ def test_get_clock_info(self):
             clocks.append('monotonic')
 
         for name in clocks:
-            info = time.get_clock_info(name)
+            with warnings.catch_warnings():
+                warnings.filterwarnings("ignore", "", DeprecationWarning)
```

Comment:
Add a comment, otherwise we can forgot why this was added.

But I prefer moving all the code from the loop body to a local function and calling it multiple times.
```python
def test(name):
    ...
test('perf_counter')
test('process_time')
test('time')
if hasattr(time, 'monotonic'):
    test('monotonic')
with self.assertWarns(DeprecationWarning):
    test('clock')
```
This silences a warning only for clock, tests that a warning for clock is raised, and in case of test failure the traceback contains a line with a timer's name.

Reply:
I changed the code to make the warning check explicit.

Moreover, hasattr(time, 'monotonic') test is outdated: time.monotonic() is always available since Python 3.5. I also fixed that.